### ISSUE TITLE
Provide a way to do lazy-loading of currencies

### ISSUE THREAD

**Issue by giosh94mhz** [mathiasverraes/money#94]
The library is quite useful and can be integrated seamlessly in many domain.

The major issue I found is that the library rely on a `currencies.php` file, which load all the currencies. This, IMO have two drawbacks:

* currencies cannot be lazy loaded, and therefore
* currencies cannot be easily stored on database (or equivalent).

I think it would be better to return a simple object with a "load" method, similar to what is done by `composer`.

e.g.

```php
// CurrencyLoader.php
interface CurrencyLoader {
    public function load($name);
//    public function loadByIso($iso);
}

// currencies.php
final class DefaultCurrencyLoader {
    private static $currencies = array(
        "AED" => "United Arab Emirates Dirham",
        "AFN" => "Afghan Afghani",
        // many other
    );

    public function load($name)
    {
         if (! isset(self::$currencies[$name])
            throw new UnknownCurrencyException();
         return self::$currencies[$name];
    }
);
```

I know that `Money` are threated as final immutable classes, and I agree, but currencies are a bit different IMO.

What do you think? Should I open a PR? :)

**Comment by giosh94mhz** [comment]
I just saw that in PR #58 you've added a class `ISOCurrencies` which implements `AvailableCurrencies`. I think, this interface can be a good starting point for what I wrote before...

**Comment by giosh94mhz** [comment]
Closed, just like #95
